### PR TITLE
Add token block data

### PR DESCRIPTION
**Pull request by mattinsler**
Related to https://github.com/Uniswap/mobile/issues/80.

This PR adds the type and implementation for tracking tokens at a block level. It is basically a copy/paste of the TokenHourData implementation, but with the block number and a timestamp added.

### CODE REVIEW

File: src/mappings/core.ts [hunk 1]
```diff
@@ -446,9 +458,18 @@ export function handleSwap(event: SwapEvent): void {
...
   token1HourData.feesUSD = token1HourData.feesUSD.plus(feesUSD)
 
+  token1BlockData.volume = token1BlockData.volume.plus(amount1Abs)
+  token1BlockData.volumeUSD = token1BlockData.volumeUSD.plus(amountTotalUSDTracked)
+  token1BlockData.untrackedVolumeUSD = token1BlockData.untrackedVolumeUSD.plus(amountTotalUSDTracked)
+  token1BlockData.feesUSD = token1BlockData.feesUSD.plus(feesUSD)
+
   swap.save()
   token0DayData.save()
+  token0HourData.save()
```
Comment (mattinsler) [inline src/mappings/core.ts:468]: Added a save here for hourly data. It looks like this isn't being saved right now.

File: src/mappings/core.ts [hunk 2]
```diff
@@ -446,9 +458,18 @@ export function handleSwap(event: SwapEvent): void {
...
+  token0BlockData.save()
   token1DayData.save()
+  token1HourData.save()
```
Comment (mattinsler) [inline src/mappings/core.ts:471]: Added a save here for hourly data. It looks like this isn't being saved right now.

**Review by ianlapham — APPROVED** [review]
Implementation looks good. The one thing to note is that not every block will be populated for each token (only blocks where a swap/mint/burn occured involving the token). 

But this is probably okay as this is also true for hour and day data. The strategy on a service or client would be to fill in the empty days with the stagnant price (as price cant change if there were no swaps or changes to liquidity)

**Comment by mattinsler** [comment]
Closing this in favor of a different implementation.

**Pull request closed without merging**